### ISSUE TITLE
Migration UI

### ISSUE THREAD

**Issue by exezbcz** [kodadot/nft-gallery#7552]
### Tasks: 

- https://github.com/kodadot/nft-gallery/issues/7560
- https://github.com/kodadot/nft-gallery/issues/7561
- https://github.com/kodadot/nft-gallery/issues/7562
- https://github.com/kodadot/nft-gallery/issues/7563

![image](https://github.com/kodadot/nft-gallery/assets/90852205/6ed5ea84-e313-4e79-86fb-7ada2d054a07)

# Abstract
- design available in handoff file: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=4218%3A2062&mode=design&t=VM0ypGEgcdUaSoT1-1
![image](https://github.com/kodadot/nft-gallery/assets/90852205/a1075c5e-7bd6-4626-b5b2-7950a5278767)
- this ui should enable migration capabalities
- consists of 4 pages - home page, overview, signing, congratulations message
- many parts similar to the admin panel, signing modals, modals in general, drop page (cards).
- as in the admin panel issue, it's quite big, please use the developer option in Figma - you easily see the padding etc. 

### Tasks: 

- https://github.com/kodadot/nft-gallery/issues/7560
- https://github.com/kodadot/nft-gallery/issues/7561
- https://github.com/kodadot/nft-gallery/issues/7562
- https://github.com/kodadot/nft-gallery/issues/7563

Now for the design: 

------

------

------

# Home page

## Top part
<img width="785" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/6c93f71d-910b-47b6-b8d8-40f89f6e4a83">
- introduces into migration, shows important parts - like the route etc.

### source and destination
- same as in teleport v2 (#7363) 
- gives the option to choose whether migrating from Polkadot assethub or Kusama asset hub. The default is polkadot. 
<img width="350" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/00c57ad8-8a02-4788-8c49-ced4350ed7a7">
- The source is Kusama, or if its more clearer maybe rmrk2, It is up to the devs 

### Faq
- I have decided to put faq on the top there
- migration is pretty hard to understand for new users. This will hopefully ensure the users are not confused about what is happening. 
<img width="428" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/fe4eb84f-046d-400d-9a97-289cf997faeb">

- states are on the left: 
<img width="457" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/7f9a04a5-6e2f-4d0f-8cc9-8e9369914074">
- please note that the height will be taken from the card with the most text - other cards will inherit the height (that means cards are consistent and won't jump around) 
- if you would like to add anything as an explainer to the cards, feel free to comment. 

## Selection
- under the full-width divider, there is a contextual message with the account you are connected with, and then there is a little tooltip section. 
- there is also a state for when you are not connected. 
<img width="838" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/139d449d-dd57-4bd5-8351-a2f7e0bd3c78">
- tooltip copy for that as well. 
<img width="370" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/eb3aea1a-ae74-43b5-aada-0d82d65f257b">

- the selection is filled with drop landing page-like cards. 
<img width="489" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/1c412ccb-cfbb-4dec-83af-3f0b4c9419a5">

- cards have a hover effect, and the migrate button itself has one as well. 
<img width="291" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/2735cd73-a36c-45d4-857c-f7da7b605c90">

- you can try it yourself in the prototype next to the card component. 

### Ready for Migration
- Here, you will be able to view collections that you own.
- >As the creator or owner of a collection, you can initiate migration here. Keep in mind only items owned by you can be migrated at this stage. You'll need to pre-sign items owned by others to allow individual migrations.
- if there is no collection to be migrated from this part, this part is simply hidden. 

### Waiting for your action
<img width="512" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/59b9507c-f164-4d6b-b3e5-c5581d02bbe3">

- here are items from collections you don't own, but the collection was migrated and initiated - giving you the option to migrate as well. 
- >Displayed here are individual items that you own from collections that have been migrated. Ensure you're up-to-date by migrating these pieces right away. Please note you can only start the migration once the collection owner initiates it.

### migration not possible
- pretty sure there will be some collections that you won't be able to migrate. For this we have the last part under divider. 
<img width="714" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/d45afd8d-0138-45a9-8b8f-d58a6b980b9f">

------

------

------

small divider 🍡 

# Review
- you will get there by clicking any button with label "migrate...." on the cards 
- the UI divides here to migration of the mentioned collection or only the items inside. 

## Migrate collection
<img width="437" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/be07efe1-f1a4-44c1-b059-e6f2c9406077">

- top part review - sign (same as in the admin panel) 
- contextual message ( connected with)
- then there is migrate collection 
- there is picture of the collection, name and message that the name will stay the same.
- divider, and now for the harder part. 
- there is row that shows "items available for migration" 
- little message: 
   - >Please note that at this stage, only items you own can be migrated. For other items, you will need to pre-sign them to allow others to follow your migration initiative.
- If I understand correctly, you cannot migrate items you don't own. Therefore, you pre-sign for others and give them access to follow your migration once they decide to do so. 
- is that correct @vikiival ? 
- then there is route explained
<img width="565" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/c6c767c4-b462-4edc-a913-9a60a5b0bbf1">

- divider and then costs, those are also similar to the #7241 
- collapsable fee breakdown, divided into network fee and then existential deposit, and then kodadot fee (if there will be any..) 
- then a row of estimated total fees. 

### bottom part 
- here is acknowledgment before you proceed, its irreversible and you should know it. 
- disabled button till you don't check the checkbox 

## Migrate items
<img width="599" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/d908736e-8f00-4378-865f-b92ee61fd20d">
- missing available items and the info part. 
- only 2 items max are shown and then there is text of how many items are left. 
- >+ 10 more items from the same collection 

------

------

------

# Signing 🚥 
- new part, will be the same as with auto teleport (but in modal) 
- the reason everything is on its own page and not in the small modal is simply the space and no way of accidentally closing it and losing the progress. It's quite a lot of transactions if you decide to migrate 1k collection. 
<img width="373" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/c2f880ee-69e6-444f-a633-07c68cbecaf3">

- it's a step signing.
- here are possible events of the small sub-part 
<img width="572" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/f17064f3-8e42-4f4e-be46-031199365796">

- the parent section changes based on whether its in progress, its done, or inactive, changes between three states. 
<img width="119" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/e01e518a-6775-4689-8d61-4222f94aa006"> <img width="71" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/40ca3d0a-6073-45c6-92bc-f158cdc5435d"> <img width="120" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/bfbacaba-80c5-49d2-8155-74ab9fbf8515">

- the grey 1/4 circle is a loading wheel, please match the style as much as possible. 
- The same icons are used for the subsections. Every step is in this case one signed transaction. 
- let me give an example. 
![image](https://github.com/kodadot/nft-gallery/assets/90852205/16f8397b-4f07-4a91-bddf-91607fc9a3f4)

- the contextual message is in this case what is changing. 
- when you cancel signing, you have the option to retry (it could say failed or canceled) 
<img width="627" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/5da95aa4-8e6b-4ea3-8eed-d280b3f565b0">

- after you finish all of this long signing, you can click on the button to finish (it's not automatic because in this signing there are also transaction links (as you can see above) 

------

------

------

# Congratulations 
- not only you have made it to the end of this issue, but the user successfully signed all the transactions and migrated. 

<img width="834" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/653f7dea-d52c-4fda-9af1-256a9a28a930">

- graphics, copy, and buttons to view the collection and to share on Twitter. Copy for the tweet can be done ad hoc. 
- I will create the graphics for dark mode as well, don't worry! 

### End 
- if you have any questions, feel free to ask

**Comment by exezbcz** [comment]
I will probably divide it into smaller tasks later today.

**Comment by vikiival** [comment]
> If I understand correctly, you cannot migrate items you don't own. Therefore, you pre-sign for others and give them access to follow your migration once they decide to do so.
is that correct @vikiival ?

Yup just checking with @jsidorenko if pre-signatures can be made in batch or you need to click 200 times sign. 
In the later option I would opt in for other solution

**Comment by exezbcz** [comment]
ready guys 👀 cc @kodadot/internal-dev 

### Tasks: 

- https://github.com/kodadot/nft-gallery/issues/7560
- https://github.com/kodadot/nft-gallery/issues/7561
- https://github.com/kodadot/nft-gallery/issues/7562
- https://github.com/kodadot/nft-gallery/issues/7563

**Comment by STUDIOUS-WOLF** [comment]
is this issue in hacktober fest if yes I would like to complete some tasks?

**Comment by exezbcz** [comment]
@STUDIOUS-WOLF Hi! Unfortunately, no. However, you could check for other issues in the repository. The frontend for this issue is mostly finished. We are now waiting for the backend afaik

**Comment by vikiival** [comment]
> We are now waiting for the backend afaik

Backend is done sir 🥹. 
Everything tested on friday. 
Cc @preschian

**Comment by exezbcz** [comment]
> Backend is done sir 🥹.

letsgoooo
yeah, that is why i put the "afaik" :D

**Comment by yangwao** [comment]
so meanwhile done 👀

**Comment by preschian** [comment]
> so meanwhile done 👀

oh, small things, there is no menu/button to migrate page

**Comment by yangwao** [comment]
- #7560 done as well?

**Comment by preschian** [comment]
> * [Migration UI - 1 - Home page  #7560](https://github.com/kodadot/nft-gallery/issues/7560) done as well?

oh, leftover related to #8197. I will pick that up

**Comment by preschian** [comment]
followup issue: https://github.com/kodadot/nft-gallery/issues/8206

**Comment by prury** [comment]
I think once we have the public working version of migration, we should ask everyone on the team to test it at least once, I'm specially concerned about the possibility of someone burning their NFTs on the source chain and not being able to mint them on the next one.

**Comment by prury** [comment]
> > I'm specially concerned about the possibility of someone burning their NFTs on the source chain
> 
> good point, should we disable auto burn? because once burned we can revert that right? to reduce the risk
> 
> replace burn section with info that tell the user don't forget to burn the old collections?

this or a button somewhere after they had minted the new items, but with disclaimer asking user to verify if new collection have been properly minted on new chain

or make burning items the last step, with link to the new collection asking user to verify if items were properly minted

**Comment by vikiival** [comment]
> or make burning items the last step, with link to the new collection asking user to verify if items were properly minted

Yup burning should be the last step. It should verify that it was minted on the destination chain.

**Comment by preschian** [comment]
> this or a button somewhere after they had minted the new items, but with disclaimer asking user to verify if new collection have been properly minted on new chain
>
> or make burning items the last step, with link to the new collection asking user to verify if items were properly minted

@exezbcz @prury just want to make sure. Regarding this, we need to adjust some UI, right? On Canary, I already put the `burn` on the last step before going to the congrats page. The easiest solution is to put the button link to the old collection on the congrats page. And user can burn on that page, since we had this button on collection page

![image](https://github.com/kodadot/nft-gallery/assets/734428/ef0d5055-ece4-4d0e-9676-7f8ad073d12e)

**Comment by roiLeo** [comment]
I think burn/destroy collection for rmrk2 is not yet implemented

**Comment by vikiival** [comment]
> I think burn/destroy collection for rmrk2 is not yet implemented

Any blocker there?

**Comment by yangwao** [comment]
Regarding this, what is outstanding for final burn? 

https://github.com/kodadot/nft-gallery/pull/8280#issuecomment-1835985956

**Comment by preschian** [comment]
> Regarding this, what is outstanding for final burn?
> 
> [#8280 (comment)](https://github.com/kodadot/nft-gallery/pull/8280#issuecomment-1835985956)

> round of team testing

need help everyone to test the features on Canary https://canary.kodadot.xyz/migrate. don't forget to open the console while testing
cc @kodadot/internal @kodadot/internal-dev 

> open design issue to define final burning phase where the user verifies if content was burned
> PR for burning

Can we put this on the congratulations page? So it won't block the testing. on the first phase of testing, save URL on the congrats page, once `PR for burning` is merged, visit the URL again to test the `burning` parts

**Comment by prury** [comment]
> Regarding this, what is outstanding for final burn?
> 
> [#8280 (comment)](https://github.com/kodadot/nft-gallery/pull/8280#issuecomment-1835985956)

i think @exezbcz and @vikiival were discussing the final burning part

**Comment by vikiival** [comment]
> Regarding this, what is outstanding for final burn?

With @exezbcz we iterated to the point that we can leverage the beauty of immutable data -> 
Therefore:

We fetch nfts from both chains.
Check if metadata hashes are matching -> as they should because it is migration.
for those who match we can safely burn them .

**Comment by exezbcz** [comment]
> @exezbcz, any updates on this?

Sorry for the delay! I jIt's for the burning part, so it's not critical. :D

**Comment by exezbcz** [comment]
continue to -> 
- https://github.com/kodadot/nft-gallery/issues/8293


<img src="https://media1.giphy.com/media/XnObKxfjNrSkU/giphy.gif"/>
